server/server.go

```diff
@@ -885,19 +867,98 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		rawGithubClient,
 	)
 
+	ctx, _ := context.WithTimeout(context.Background(), 5*time.Second) // nolint: vet
+	gatewaySnsWriter := sns.NewWriterWithStats(session, userConfig.LyftGatewaySnsTopicArn, statsScope)
+	autoplanValidator := &gateway.AutoplanValidator{
+		Logger:                        logger,
+		Scope:                         statsScope.SubScope("validator"),
+		VCSClient:                     vcsClient,
+		PreWorkflowHooksCommandRunner: preWorkflowHooksCommandRunner,
+		Drainer:                       drainer,
+		GlobalCfg:                     globalCfg,
+		AllowForkPRs:                  userConfig.AllowForkPRs,
+		AllowForkPRsFlag:              config.AllowForkPRsFlag,
+		SilenceForkPRErrors:           userConfig.SilenceForkPRErrors,
+		SilenceForkPRErrorsFlag:       config.SilenceForkPRErrorsFlag,
+		CommitStatusUpdater:           commitStatusUpdater,
+		PrjCmdBuilder:                 projectCommandBuilder,
+		PullUpdater:                   pullUpdater,
+	}
+	gatewayEventsController := &gateway.VCSEventsController{
```

Comment:
admittedly, I can move the creation of these structs into the Gateway and Hybrid modes, I just wanted to keep the switch readable/avoid extracting out a helper fxn with so many parameters to generate them.

Reply:
Yeah I agree it's a lot of parameters but in my opinion, it would definitely help with the readability. @nishkrishnan  and @msarvar  would know better :)